File: packages/flutter_tools/lib/src/base/clock.dart
// Copyright 2017 The Chromium Authors. All rights reserved.
// Use of this source code is governed by a BSD-style license that can be
// found in the LICENSE file.

import 'package:quiver/time.dart';

import 'context.dart';

const Clock _systemClock = const Clock();

/// Currently active clock implementation.
///
/// By default uses system clock. Override this in tests using [Clock.fixed].
Clock get clock {
  return context == null
      ? _systemClock
      : context.putIfAbsent(Clock, () => _systemClock);
}

File: engine/src/flutter/shell/platform/fuchsia/dart-pkg/zircon_ffi/clock.cc
// Copyright 2013 The Flutter Authors. All rights reserved.
// Use of this source code is governed by a BSD-style license that can be
// found in the LICENSE file.

#include "clock.h"

#include <zircon/syscalls.h>

uint64_t zircon_dart_clock_get_monotonic() {
  return zx_clock_get_monotonic();
}
